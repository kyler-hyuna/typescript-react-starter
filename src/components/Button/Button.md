Button example:

```js
<Button title="I'm Cool">Push Me</Button>
```
