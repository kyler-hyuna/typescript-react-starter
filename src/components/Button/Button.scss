.button {
  background-color: rebeccapurple;
  color: #fff;
}
