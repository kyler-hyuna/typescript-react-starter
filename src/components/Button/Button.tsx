import React from 'react'
import './Button.scss'

export interface ButtonProps {
  /** Title of the button */
  title: string
}

/**
 * Button
 */
export const Button: React.SFC<ButtonProps> = ({ title }) => (
  <button className="button">{title}</button>
)

export default Button
