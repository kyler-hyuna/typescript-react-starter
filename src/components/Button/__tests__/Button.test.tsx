import React from 'react'
import { create } from 'react-test-renderer'
import Button from '../Button'

describe('Button', () => {
  it('should work', () => {
    const tree = create(<Button title="I'm a button" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
