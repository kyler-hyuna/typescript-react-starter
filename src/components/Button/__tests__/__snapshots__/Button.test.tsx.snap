// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should work 1`] = `
<button
  className="button"
>
  I'm a button
</button>
`;
